### PR TITLE
docs: Add embedding explanation

### PR DESCRIPTION
**Pull request by TomChv**
Resolves: #2042

Signed-off-by: Vasek - Tom C <tom.chauveau@epitech.eu>

### CODE REVIEW

**Review by gerhard — COMMENTED** [review]
- [x] Can we revert the truncation changes to the lines which do not change? It's difficult to see what changed if those get auto-formatted.
- [x] Would it be possible to continue with the same theme as we had before, i.e. Bob / Engineer / Person?
- [ ] Could we add a link to the CUE playground sample, same as we have for the other CUE snippets? e.g. [Try it in the CUE playground](https://tip.cuelang.org/play/?id=_Cvwm6KeGZm#cue@export@yaml)
- [ ] A few improvements to the first two sentences:
```diff
- In the same aspect of unification, CUE can embed definition into another to merge it. It's not much different from [Golang Embedding](https://go.dev/doc/effective_go#embedding).
- It's a powerful feature to extends definition's field without adding a level of depth in the top definition.
+ CUE allows the embedding of one definition into another, similar to [Golang Embedding](https://go.dev/doc/effective_go#embedding) or [object-oriented composition](https://en.wikipedia.org/wiki/Composition_over_inheritance). This avoids adding an extra level of depth to the definition.
```

**Comment by dolanor** [comment]
Agree with the reformatting, hard to see what's changed.

File: docs/core-concepts/1215-what-is-cue.md [hunk 1]
```diff
@@ -206,6 +207,60 @@ If we wanted the Bob object to have a different job, it would either need to be
...
+// output in YAML:
+Bob:
+  Name: Bob Smith
+  Email: bob@smith.com
+  Age: 42
+  Job: Engineer
+  Domain: Backend
+```
+
+Any definition that embed `#Engineer` will share his properties, and they will be accessible directly from the definition.
```
Comment (dolanor) [inline docs/core-concepts/1215-what-is-cue.md:262]:
```suggestion
Any definition that embed `#Engineer` will share its properties, and they will be accessible directly from the definition.
```

Comment (TomChv) [reply docs/core-concepts/1215-what-is-cue.md]: Fixed in https://github.com/dagger/dagger/pull/2046/commits/7cbb4639c53b0521f408ccc567a88cd13c7844b7

**Review by gerhard — APPROVED** [review]
@TomChv there is one item left (see my comment above), but we are going to merge this as is, it's OK to follow-up later if you want. Thank you!

**Pull request merged**